### ISSUE TITLE
[lc_ctrl] D2 Signoff

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#20979]
### Description

Ensure D2 signoff criteria are fulfilled after focus area changes have landed.

**Comment by msfschaffner** [comment]
Changes since ES tapeout include misc changes in docs, general reggen improvements:

```
$ git log Earlgrey-M2.5.2-RC0..HEAD --oneline hw/ip/lc_ctrl

fa447d0cdc [lc_ctrl] Route Flash RMA ACK signals in parallel
85496723e4 [lc_ctrl] Add OTP state encoding for auth states
41c053e8da [lc_ctrl] Add OTP state encoding for ownership states
ce422750fe [lc_ctrl] Always allow transitions into scrap
3a8dd54137 [lc_ctrl] Bump version and revert to D1/V1
61a237e197 [util/reggen] reverse order of substruct generation
c939d9a6d5 [otp_ctrl] Add support for multiple HW_CFG partitions
ce648ca68e [ipgen.pwrmgr] Change core files to vlnv naming and label as virtual
de31bdf1c2 [reggen] Remove the devmode input
637d1b88e1 [doc,reset] Fix references to reset behavior
3fe399b6fe [lc_ctrl/sw] Correct ID state enum
3f88a55120 [pwrmgr,ipgen] Generate pwrmgr ip_autogen files with topgen
1b16ca2122 [reggen] Add mubi support SWAccess that sets/clears a reg
03f475100d [doc,lc_ctrl] Fix incorrect link
d9baa965dd [SiVal] Test plan updates for `lc_ctrl`
59f8142826 [doc] Moved badges over to using hosted images
97f9a3c7ec [doc] lc_ctrl registers and interfaces now use CMDGEN
2d61350be6 Integrate DRG3 class for generating randomness
7688e714e8 [reggen] Add initial support for version and cip_id hjson fields
fbd888eea8 Revert "[reggen] Add CIP_IDs and bump all major versions"
0ba10b3cd3 [reggen] Add CIP_IDs and bump all major versions
82ac1e9358 [lc_ctrl] Fix CDC bug in lc_ctrl kmac interface
```

The relevant changes in this block, impacting RTL are the following three:
1. fa447d0cdc [lc_ctrl] Route Flash RMA ACK signals in parallel -> this makes the design more secure
1. ce422750fe [lc_ctrl] Always allow transitions into scrap -> this was a bug
1. 82ac1e9358 [lc_ctrl] Fix CDC bug in lc_ctrl kmac interface -> this was a bug

Changes 1.&2. are modeled in DV at the block-level.
3. has been found with GLS and will be retested in GLS later towards the TO.

No TODOs have been added to RTL.
A DV TODO has been added for V3 coverage closure: #21268

The block should hence still fulfill the D2S criteria.

**Comment by msfschaffner** [comment]
@vogelpi @andreaskurth can you take a look at this D2S signoff analysis? thanks!

**Comment by andreaskurth** [comment]
I agree that D2S is still fulfilled. I reviewed the following commits, which change `lc_ctrl`'s RTL code, in detail (with `git log -p Earlgrey-M2.5.2-RC0..HEAD -- hw/ip/lc_ctrl/rtl`):
- https://github.com/lowRISC/opentitan/commit/60650e073fa706966d8277fe7b55fdaac806a807 [lc_ctrl] Use unique diversification values
- https://github.com/lowRISC/opentitan/commit/fa447d0cdc993726850ef4422a32e473434a6ba0 [lc_ctrl] Route Flash RMA ACK signals in parallel
- https://github.com/lowRISC/opentitan/commit/85496723e496f19c20a2477d3d4a31090bde19e8 [lc_ctrl] Add OTP state encoding for auth states
- https://github.com/lowRISC/opentitan/commit/41c053e8da07bb47d02b792108a9bcf23a949425 [lc_ctrl] Add OTP state encoding for ownership states
- https://github.com/lowRISC/opentitan/commit/ce422750fe05117e511f17dfc1efb4051c142ca3 [lc_ctrl] Always allow transitions into scrap
- https://github.com/lowRISC/opentitan/commit/61a237e1973d6a29d377c2be871139553c7d344f [util/reggen] reverse order of substruct generation
- https://github.com/lowRISC/opentitan/commit/de31bdf1c26be2714ca787c6041a56eb56eb7d1e [reggen] Remove the devmode input
- https://github.com/lowRISC/opentitan/commit/1b16ca2122d956f4b1f909ba2e996b01ff22001e [reggen] Add mubi support SWAccess that sets/clears a reg
- https://github.com/lowRISC/opentitan/commit/2d61350be65bdbd02d23ed28f4782cae2662962b Integrate DRG3 class for generating randomness
- https://github.com/lowRISC/opentitan/commit/fbd888eea85260f03095b7b7f12537638ca27183 Revert "[reggen] Add CIP_IDs and bump all major versions"
- https://github.com/lowRISC/opentitan/commit/0ba10b3cd3157d233c6a9a21676b058971399b99 [reggen] Add CIP_IDs and bump all major versions
- https://github.com/lowRISC/opentitan/commit/82ac1e93582708d68680043f9f4edfb25240718d [lc_ctrl] Fix CDC bug in lc_ctrl kmac interface

**Comment by vogelpi** [comment]
I've previously reviewed all the important RTL changes to lc_ctrl and agree that the block still meets the D2S criteria.

With @andreaskurth , I've discussed that it's sufficient to follow the flow for blocks with minor changes. I am now closing the issue and will merge the PR bringing lc_ctrl back to D2S. Thanks @msfschaffner for the detailed analysis and for putting this together, thanks @andreaskurth for your review.
